### PR TITLE
3708 address historical federal awards with missing additional award identification

### PR DESCRIPTION
**Pull request by sambodeme**
## Description
Some historical reports failed the first round of migration due to missing additional award identification. Upon investigation, it was discovered that the failure was due to invalid characters associated with the additional identification (value contains =""516084" instead of 516084) in the audit.AWARDIDENTIFICATION field in the database. This prevented the code from properly generating the awards workbook and hence from properly validating the data. The solution applied here was to add logic that sanitizes the audit.AWARDIDENTIFICATION values by removing leading `=""` and trailing `" ` for Excel.

## How to test
1. Launch the stack from the main branch.
2. Identify a valid report with non empty `audit.AWARDIDENTIFICATION`  values.
3. Change any `audit.AWARDIDENTIFICATION`  field value by adding `=""` and trailing `"` (for instance, change `123` to `=""123"``.
4. Attempt to migrate this report and notice how it fails due to "missing additional award identification"
5. Switch to this PR branch and re-run the migration for the same report and observe that it processes properly.

For extra points, you may try to create from scratch an audit report and submit it from this PR branch to confirm that process works as expected.

## PR checklist: submitters

- [ ]   Link to an issue if possible. If there’s no issue, describe what your branch does. Even if there is an issue, a brief description in the PR is still useful.
- [ ]   List any special steps reviewers have to follow to test the PR. For example, adding a local environment variable, creating a local test file, etc.
- [ ]   For extra credit, submit a screen recording like [this one](https://github.com/GSA-TTS/FAC/pull/1821).
- [ ]   Make sure you’ve merged `main` into your branch shortly before creating the PR. (You should also be merging `main` into your branch regularly during development.)
- [ ]   Make sure you’ve accounted for any migrations. When you’re about to create the PR, bring up the application locally and then run `git status | grep migrations`. If there are any results, you probably need to add them to the branch for the PR. Your PR should have only **one** new migration file for each of the component apps, except in rare circumstances; you may need to delete some and re-run `python manage.py makemigrations` to reduce the number to one. (Also, unless in exceptional circumstances, your PR should not delete any migration files.)
- [ ]   Make sure that whatever feature you’re adding has tests that cover the feature. This includes test coverage to make sure that the previous workflow still works, if applicable.
- [ ]   Make sure the full-submission.cy.js [Cypress test](https://github.com/GSA-TTS/FAC/blob/main/docs/testing.md#end-to-end-testing) passes, if applicable.
- [ ]   Do manual testing locally. Our tests are not good enough yet to allow us to skip this step. If that’s not applicable for some reason, check this box.
- [ ]   Verify that no Git surgery was necessary, or, if it was necessary at any point, repeat the testing after it’s finished.
- [ ]   Once a PR is merged, keep an eye on it until it’s deployed to dev, and do enough testing on dev to verify that it deployed successfully, the feature works as expected, and the happy path for the broad feature area (such as submission) still works.

## PR checklist: reviewers

- [ ]   Pull the branch to your local environment and run `make docker-clean; make docker-first-run && docker compose up`; then run `docker compose exec web /bin/bash -c "python manage.py test"`
- [ ]   Manually test out the changes locally, or check this box to verify that it wasn’t applicable in this case.
- [ ]   Check that the PR has appropriate tests. Look out for changes in HTML/JS/JSON Schema logic that may need to be captured in Python tests even though the logic isn’t in Python.
- [ ]   Verify that no Git surgery is necessary at any point (such as during a merge party), or, if it was, repeat the testing after it’s finished.

The larger the PR, the stricter we should be about these points.

### CODE REVIEW

File: backend/census_historical_migration/workbooklib/federal_awards.py [hunk 1]
```diff
@@ -544,6 +552,40 @@ def xform_populate_default_loan_balance(audits):
     return loans_at_end
 
 
+def xform_sanitize_additional_award_identification(audits, identifications):
+    """Sanitize the input to ensure it does not start with ="" and end with " which might be interpreted as a formula in Excel."""
+
+    change_records = []
+    news_identifications = []
```
Comment (phildominguez-gsa) [inline backend/census_historical_migration/workbooklib/federal_awards.py:559]:
```suggestion
    new_identifications = []
```
Typo?

File: backend/census_historical_migration/workbooklib/federal_awards.py [hunk 2]
```diff
@@ -544,6 +552,40 @@ def xform_populate_default_loan_balance(audits):
...
+    has_modified_identification = False
+    for audit, identification in zip(audits, identifications):
+        if (
+            identification
+            and identification.startswith('=""')
+            and identification.endswith('"')
+        ):
+
+            new_identification = identification[3:-1]
```
Comment (phildominguez-gsa) [inline backend/census_historical_migration/workbooklib/federal_awards.py:568]:
```suggestion
        ):
            new_identification = identification[3:-1]
```

File: backend/census_historical_migration/workbooklib/federal_awards.py [hunk 3]
```diff
@@ -544,6 +552,39 @@ def xform_populate_default_loan_balance(audits):
...
+        if (
+            identification
+            and identification.startswith('=""')
+            and identification.endswith('"')
+        ):
+            new_identification = identification[3:-1]
+            has_modified_identification = True
+        else:
+            new_identification = identification
+        news_identifications.append(new_identification)
```
Comment (sambodeme) [inline backend/census_historical_migration/workbooklib/federal_awards.py:571]:
```suggestion
        new_identifications.append(new_identification)
```

File: backend/census_historical_migration/workbooklib/federal_awards.py [hunk 4]
```diff
@@ -544,6 +552,39 @@ def xform_populate_default_loan_balance(audits):
...
+            "additional_award_identification",
+            new_identification,
+            ["xform_sanitize_additional_award_identification"],
+            change_records,
+        )
+    # See Transformation Method Change Recording at the top of this file.
+    if change_records and has_modified_identification:
+        InspectionRecord.append_federal_awards_changes(change_records)
+
+    return news_identifications
```
Comment (sambodeme) [inline backend/census_historical_migration/workbooklib/federal_awards.py:585]:
```suggestion
    return new_identifications
```

**Review by phildominguez-gsa — DISMISSED** [review]

**Review by phildominguez-gsa — APPROVED** [review]

**Review by phildominguez-gsa — APPROVED** [review]

**Pull request merged**